src/visualization/parsers/boxmodel.ts

```diff
@@ -0,0 +1,186 @@
+/// <reference path="../../../node_modules/clarity-js/clarity.d.ts" />
+import { IParser } from "../components/Snapshot";
+
+export default class BoxModel implements IParser {
+    private layouts: { [index: number]: Node } = {};
+    private states: { [index: number]: IElementLayoutState } = {};
+    private document: Document;
+    private base: string;
+   
+    setup(document: Document, frame: HTMLIFrameElement, base: string, thumbnail? : boolean) {
+        this.layouts = {};
+        this.document = document;
+        this.base = base;
+    }
+
+    private domInsert(node: Node, parent?: Node, nextSibling?: Node) {
```

Comment:
>parent [](start = 34, length = 6)

is this actually optional? looks like domInsert is a noop if it isn't passed

Reply:
I'll make it required, thanks.